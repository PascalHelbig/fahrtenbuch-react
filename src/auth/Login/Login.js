import React, { PropTypes } from 'react';
import RaisedButton from 'material-ui/RaisedButton/RaisedButton';
import Header from '../../Header/Header';

const Login = ({ onLogin }) => (
  <div>
    <Header>Login</Header>
    <RaisedButton
      label="login"
      onTouchTap={() => onLogin('test@test.de', '12345')}
      primary
    />
  </div>
);

Login.propTypes = {
  onLogin: PropTypes.func.isRequired,
};

export default Login;
