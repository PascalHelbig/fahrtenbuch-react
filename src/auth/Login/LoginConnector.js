import React, { PropTypes } from 'react';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import Login from './Login';

export const LoginConnectorDumb = ({ login }) => (
  <Login
    onLogin={(email, password) => login({ variables: { email, password } })}
  />
);

LoginConnectorDumb.propTypes = {
  login: PropTypes.func.isRequired,
};

const loginMutation = gql`mutation login($email: Email!, $password: Password!) {
  login( email: $email, password: $password) {
    token
  }
}`;

export default graphql(loginMutation, { name: 'login' })(LoginConnectorDumb);
