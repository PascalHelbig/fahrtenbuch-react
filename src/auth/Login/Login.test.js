import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RaisedButton from 'material-ui/RaisedButton/RaisedButton';
import Login from './Login';

it('renders correctly', () => {
  const wrapper = shallow(<Login onLogin={() => {}} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('runs login on button touchTap', () => {
  const onLogin = jest.fn();
  const wrapper = shallow(<Login onLogin={onLogin} />);
  wrapper.find(RaisedButton).simulate('touchTap');
  expect(onLogin).toHaveBeenCalled();
});
