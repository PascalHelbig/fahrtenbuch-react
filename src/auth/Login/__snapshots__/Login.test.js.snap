exports[`test renders correctly 1`] = `
<div>
  <Header>
    Login
  </Header>
  <RaisedButton
    disabled={false}
    fullWidth={false}
    label="login"
    labelPosition="after"
    onTouchTap={[Function]}
    primary={true}
    secondary={false} />
</div>
`;
